File: backend/app/schemas/disease.py
from pydantic import BaseModel
from typing import List, Optional, Dict, Any

class AlternativePrediction(BaseModel):
    disease: str
    confidence: float

class RecoveryMilestone(BaseModel):
    phase: str
    timeline: str
    action: str

class DiseasePredictionResponse(BaseModel):
    plant: str = "Rice / Paddy"
    disease: str
    common_name: Optional[str] = None
    pathogen: Optional[str] = None
    pathogen_type: Optional[str] = "Fungal"
    confidence: float
    severity: str
    spread_risk: Optional[str] = "Medium"
    spread_risk_score: Optional[int] = 50
    symptoms: List[str] = []
    immediate_actions: Optional[List[str]] = []
    recommended_action: List[str] = []
    products: List[str] = []
    organic_remedies: Optional[List[str]] = []
    chemical_remedies: Optional[List[str]] = []
    prevention: List[str] = []
    resistant_varieties: Optional[List[str]] = []
    recovery_milestones: Optional[List[RecoveryMilestone]] = []
    top_predictions: Optional[List[AlternativePrediction]] = []
    ai_solution: Optional[str] = None
    model_source: Optional[str] = None
    disclaimer: str = "This is an AI-powered diagnostic advisory based on deep learning visual analysis. Confirm critical diagnoses with your regional agricultural extension officer or agronomist before initiating broad chemical applications."

File: frontend/src/types/disease.ts
export interface AlternativePrediction {
  disease: string;
  confidence: number;
}

export interface RecoveryMilestone {
  phase: string;
  timeline: string;
  action: string;
}

export interface DiseasePredictionResponse {
  plant: string;
  disease: string;
  common_name?: string;
  pathogen?: string;
  pathogen_type?: string;
  confidence: number;
  severity: string;
  spread_risk?: string;
  spread_risk_score?: number;
  symptoms: string[];
  immediate_actions?: string[];
  recommended_action: string[];
  products?: string[];
  organic_remedies?: string[];
  chemical_remedies?: string[];
  prevention: string[];
  resistant_varieties?: string[];
  recovery_milestones?: RecoveryMilestone[];
  top_predictions?: AlternativePrediction[];
  ai_solution?: string;
  model_source?: string;
  disclaimer: string;
}

export interface DiseaseResult {
  diseaseName: string;
  confidence: number;
  treatment: string[];
  prevention: string[];
}
